agent/api/container.go

```diff
@@ -216,6 +221,45 @@ func (c *Container) GetKnownExitCode() *int {
 	return c.knownExitCode
 }
 
+// SetRegistryAuthCredentials will set the credentials for pulling image from ecr
+func (c *Container) SetRegistryAuthCredentials(credential *credentials.IAMRoleCredentials) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.RegistryAuthentication == nil || c.RegistryAuthentication.ECRAuthData == nil {
+		return fmt.Errorf("Container: set ecr pull credentials failed, no registry auth data for container")
+	}
+
+	if !c.RegistryAuthentication.ECRAuthData.UseExecutionRole {
+		return fmt.Errorf("Container: set ecr pull credentials failed, using task level credentials is disabled")
+	}
+
+	c.RegistryAuthentication.ECRAuthData.PullCredentials = *credential
```

Comment:
I don't think you need to pass a pointer for `*credentials.IAMRoleCredentials`. It can just be `SetRegistryAuthCredentials(credential credentials.IAMRoleCredentials)`.